services/src/main/java/org/fao/geonet/api/users/UsersApi.java

```diff
@@ -226,8 +228,17 @@ public User getUser(
         // elsewhere in the GeoNetwork database - an exception is thrown if
         // this is the case
         if (dataManager.isUserMetadataOwner(userIdentifier)) {
+            MetadataRepository metadataRepository = ApplicationContextHolder.get().getBean(MetadataRepository.class);
+            final List<Metadata> allUserRecords = metadataRepository.findAll(MetadataSpecs.isOwnedByUser(userIdentifier));
+            List<String> message = new ArrayList<>();
+            for (Metadata record : allUserRecords) {
```

Comment:
What about if the user owns for example 10000 records?

Reply:
The list of records is available in the user page where the delete action is. So not sure it will bring much benefit. Basically the issue we were fixing here for geocat.ch is that subtemplate were not listed and admin user was not able to understand why the user can't be deleted. So I can remove the list of records from the error message and let the client take care of in case of error, do something like list all user records ?